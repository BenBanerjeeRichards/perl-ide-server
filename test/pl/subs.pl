sub signatureNoAttribute($a, $b = 54, $c=5) {
    say "f(...) " . $a;
    say ($b);
}

# #Prototype, no attributes
sub protoNoAttribute($$@) {
}

# Signature with attributes
sub signatureWithAttribute:lvalue(a,b) method :  test(34, $fv, tr) ($a, $b)     {
}

sub protoWithAttribute($$@[$%]) :lvalue {

}

sub f : attr('()', '(', ')', '\(') {
}
