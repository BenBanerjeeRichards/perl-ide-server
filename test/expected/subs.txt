Sub
SubName(signatureNoAttribute)
Signature(($a, $b = 54, $c=5))
LBracket
Builtin(say)
StringStart(")
String(f(...) )
StringEnd(")
Dot
ScalarVariable($a)
;
Builtin(say)
LParen
ScalarVariable($b)
RParen
;
RBracket
Comment(# #Prototype, no attributes)
Sub
SubName(protoNoAttribute)
Prototype(($$@))
LBracket
RBracket
Comment(# Signature with attributes)
Sub
SubName(signatureWithAttribute)
AttributeColon(:)
Attribute(lvalue)
AttributeArgs((a,b))
Attribute(method)
AttributeColon(:)
Attribute(test)
AttributeArgs((34, $fv, tr))
Signature(($a, $b))
LBracket
RBracket
Sub
SubName(protoWithAttribute)
Prototype(($$@[$%]))
AttributeColon(:)
Attribute(lvalue)
LBracket
RBracket
Sub
SubName(f)
AttributeColon(:)
Attribute(attr)
AttributeArgs(('()', '(', ')', '\('))
LBracket
RBracket
EndOfInput
